src/sentry/region_to_control/consumer.py

```diff
@@ -70,25 +71,49 @@ def flush_batch(self, batch: Sequence[RegionToControlMessage]):
     def _flush_batch(self, batch: Sequence[RegionToControlMessage]):
         for row in batch:
             if row.user_ip_event:
-                updated, created = UserIP.objects.create_or_update(
-                    values=dict(
-                        user_id=row.user_ip_event.user_id,
-                        ip_address=row.user_ip_event.ip_address,
-                        last_seen=row.user_ip_event.last_seen,
-                        country_code=row.user_ip_event.country_code,
-                        region_code=row.user_ip_event.region_code,
-                    )
-                )
-
-                if created:
-                    metrics.incr("region_to_control.consumer.user_ip_event.created")
-                elif updated:
-                    metrics.incr("region_to_control.consumer.user_ip_event.updated", amount=updated)
-                else:
-                    # This happens when there is an integrity error adding the UserIP -- such as when user is deleted,
-                    # or the ip address does not match the db validation.  This is expected and not an error condition
-                    # in low quantities.
-                    metrics.incr("region_to_control.consumer.user_ip_event.stale_event")
+                self._handle_user_ip_event(row.user_ip_event)
+            if row.audit_log_event:
+                self._handle_audit_log_event(row.audit_log_event)
+
+    def _handle_audit_log_event(self, audit_log_entry: AuditLogEvent, reentry=False):
+        entry = AuditLogEntry.from_event(audit_log_entry)
+        try:
+            entry.save()
+            metrics.incr("region_to_control.consumer.audit_log_entry.created")
+        except IntegrityError as e:
+            error_message = str(e)
```

Comment:
it might be useful to capture the exception and send it to Sentry. `logging.exception("Unable to save. Or stale event.")` is one way to do this.

Reply:
rgr rgr.  Silly question; when should I use our metrics tooling vs sending events?

Reply:
@corps I think having both is 👍 . I don't think it's harmful to "over"-monitor this critical developing infrastructure of hybrid cloud.

Reply:
Added, thanks for the tips!